File: Homework3/Question2/src/Cage.java
public class Cage {
    private Bird bird;

    public Cage(Bird bird) {
        this.bird = bird;
    }

    public Bird getBird() {
        return bird;
    }

    public static void testCage() {
        HoodedPitohui h1 = new HoodedPitohui();
        Cage cage = new Cage(h1);
        HoodedPitohui h2 = (HoodedPitohui) cage.getBird();
        System.out.println(h1 == h2);
    }
}

File: Homework3/Cage.kt
class Cage {
    private val bird: Bird

    constructor(bird: Bird) {
        this.bird = bird
    }

    fun getBird(): Bird {
        return bird
    }

    companion object {
        fun testCage() {
            val h1 = HoodedPitohui()
            val cage = Cage(h1)
            val h2 = cage.getBird()
            println(h1 == h2)
        }
    }
}